hyperdome_client/hyperdome_client.py

```diff
@@ -285,13 +290,35 @@ def after_start(counselor: str):
                 return
             if self.server.is_counselor:
                 self.uid = counselor
-            self.is_connected = True
-            self.get_messages_task = threads.GetMessagesTask(self.session,
-                                                             self.server,
-                                                             self.uid)
-            self.get_messages_task.setAutoDelete(False)
-            self.get_messages_task.signals.success.connect(
-                self.on_history_added)
+                @QtCore.pyqtSlot(str)
+                def counselor_got_guest(guest_key: str):
+                    if not guest_key:
+                        return
+                    self.crypt.perform_key_exchange(guest_key, self.server.is_counselor)
+                    self.poll_guest_key_task = None
+                    self.get_messages_task = threads.GetMessagesTask(self.session,
+                                                                     self.server,
+                                                                     self.uid)
+                    self.get_messages_task.setAutoDelete(False)
+                    self.get_messages_task.signals.success.connect(
+                        self.on_history_added)
+                self.poll_guest_key_task = threads.PollForConnectedGuestTask(
+                    self.session, self.server, self.uid)
+                self.poll_guest_key_task.setAutoDelete(False)
+                self.poll_guest_key_task.signals.success.connect(
+                    counselor_got_guest)
+                self.poll_guest_key_task.signals.error.connect(self.handle_error
+
```

Comment:
Weird blank line

Reply:
Trusty Ctrl-X I assume, there's a TODO there so I'll come back to it.